qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/common/data_handling/test/test_data_presenter.py

```diff
@@ -291,6 +291,82 @@ def test_handle_table_cell_changed_checkbox_ticked(self):
+        self.presenter.row_numbers = data_presenter.TwoWayRowDict()
+        self.presenter.row_numbers["TOF_WS_Name1"] = 0
+        self.presenter.row_numbers["dSpacing_WS_Name1"] = 1
+
+        ws2 = mock.MagicMock()
+        ws2.getXDimension().name = "dSpacing"
+        self.model.get_active_ws.return_value = ws2
+        self.presenter.plot_added_notifier = mock.MagicMock()
+        self.presenter.plot_removed_notifier = mock.MagicMock()
+        self.presenter.plotted.add("TOF_WS_Name1")
+        ads_patch.retrieve().getXDimension().name = "TOF"
+
+        self.presenter._handle_table_cell_changed(1, 2)  # tick the 2nd dSpacing row to be plotted
+
+        self.assertEqual(1, self.view.set_item_checkstate.call_count)
+        self.assertEqual(0, self.presenter.plot_added_notifier.notify_subscribers.call_count)
+        self.assertEqual(1, len(self.presenter.plotted))
+        self.assertEqual("TOF_WS_Name1", next(iter(self.presenter.plotted)))
+        self.view.set_item_checkstate.assert_called_once_with(1, 2, False)
+
+    @patch(dir_path + ".data_presenter.ADS")
+    def test_handle_table_cell_changed_ticked_to_plot_on_tof_when_tof_plotted(self, ads_patch):
+        self.view.get_item_checked.side_effect = _get_item_checked_mock
+        self.presenter.row_numbers = data_presenter.TwoWayRowDict()
+        self.presenter.row_numbers["TOF_WS_Name1"] = 0
+        self.presenter.row_numbers["TOF_WS_Name2"] = 1
+
+        ws2 = mock.MagicMock()
+        ws2.getXDimension().name = "TOF"
+        self.model.get_active_ws.return_value = ws2
+        self.model.get_active_ws_name.return_value = "TOF_WS_Name2"
+        self.presenter.plot_added_notifier = mock.MagicMock()
+        self.presenter.plot_removed_notifier = mock.MagicMock()
+        self.presenter.plotted.add("TOF_WS_Name1")
+        ads_patch.retrieve().getXDimension().name = "TOF"
+
+        self.presenter._handle_table_cell_changed(1, 2)  # tick the 2nd TOF row to be plotted
+
+        self.assertEqual(0, self.view.set_item_checkstate.call_count)
+        self.assertEqual(1, self.presenter.plot_added_notifier.notify_subscribers.call_count)
+        self.assertEqual(2, len(self.presenter.plotted))
+
+    @patch(dir_path + ".data_presenter.logger")
+    @patch(dir_path + ".data_presenter.ADS")
+    def test_loading_new_dspacing_file_with_ticked_to_plot_when_tof_plotted(self, ads_patch, mock_logger):
+        self.presenter.plotted = {"TOF_WS_Name1"}
+        ws1 = mock.MagicMock()
+        ws1.getXDimension().name = "TOF"
+        ws2 = mock.MagicMock()
+        ws2.getXDimension().name = "dSpacing"
+        self.model.get_sample_log_from_ws.return_value = "bankOrRunNumber"
+        self.model.get_last_added.return_value = ["dSpacing_WS_Name1"]
+        model_dict = {"TOF_WS_Name1": ws1, "dSpacing_WS_Name1": ws2}
+        self.model.get_loaded_workspaces.return_value = model_dict
+        self.view.get_add_to_plot.return_value = True
+
+        def _get_active_ws_name(name):
```

Comment:
Can these be defined as lambda functions in the `side_effect` assignment? Maybe the static analysis doesn't allow this...